src/autopas/LogicHandler.h

```diff
@@ -1120,17 +1406,26 @@ std::tuple<Configuration, std::unique_ptr<TraversalInterface>, bool> LogicHandle
 
 template <typename Particle>
 template <class Functor>
-bool LogicHandler<Particle>::iteratePairwisePipeline(Functor *functor) {
+bool LogicHandler<Particle>::computeInteractionsPipeline(Functor *functor,
+                                                         const InteractionTypeOption &interactionType) {
+  if (not _interactionTypes.count(interactionType)) {
+    autopas::utils::ExceptionHandler::exception(
+        "LogicHandler::computeInteractionsPipeline(): AutPas was not initialized for the Functor's interactions type: "
+        "{}.",
+        interactionType);
+  }
   /// Selection of configuration (tuning if necessary)
   utils::Timer tuningTimer;
   tuningTimer.start();
-  const auto [configuration, traversalPtr, stillTuning] = selectConfiguration(*functor);
+  const auto [configuration, traversalPtr, stillTuning] = selectConfiguration(*functor, interactionType);
   tuningTimer.stop();
-  _autoTuner.logIteration(configuration, stillTuning, tuningTimer.getTotalTime());
+  auto &autoTuner = _autoTunerRefs[interactionType];
+  autoTuner->logTuningResult(stillTuning, tuningTimer.getTotalTime());
+  const auto rebuildIteration = not _neighborListsAreValid.load(std::memory_order_relaxed);
```

Comment:
I just remembered why I did it this way. The value needs to be retrieved before calling `computeInteractions()`, because afterwards the neighbor lists are always valid. Added a comment now